File: shared/src/main/java/com/alta_v2/model/TiledMapDefinitionModel.java
package com.alta_v2.model;

import lombok.Builder;
import lombok.Value;

import java.util.Collections;
import java.util.List;
import java.util.Map;
import java.util.stream.Collectors;

@Value
@Builder
public class TiledMapDefinitionModel {

    private final String mapPath;
    private final String displayName;
    private final PlayerDefinitionModel player;
    private final List<NpcDefinitionModel> npcList;

    public List<Integer> getNpcIds() {
        if (npcList == null) {
            return Collections.emptyList();
        }

        return npcList.stream().map(ActorDefinitionModel::getId).collect(Collectors.toList());
    }

    public Map<Integer, String> getNpcPaths() {
        if (npcList == null) {
            return Collections.emptyMap();
        }

        return npcList.stream().collect(Collectors.toMap(ActorDefinitionModel::getId, ActorDefinitionModel::getTexturePath));
    }
}

File: shared/src/main/java/com/alta_v2/model/TiledMapDefinitionModel.kt
package com.alta_v2.model

import java.util.stream.Collectors

class TiledMapDefinitionModel(val mapPath: String,
                              val displayName: String,
                              val player: PlayerDefinitionModel,
                              val npcList: List<NpcDefinitionModel> = listOf()

) {
    val npcIds: List<Int>
        get() = npcList.stream().map(ActorDefinitionModel::id).collect(Collectors.toList())

    val npcPaths: Map<Int, String>
        get() = npcList.stream().collect(Collectors.toMap(ActorDefinitionModel::id, ActorDefinitionModel::texturePath))
}